### PR TITLE
Integration test automation

### PR DESCRIPTION
**Pull request by Erin-Mounts**
Includes initial round of automated integration tests, and fixes for a few bugs uncovered as a result; this will constitute the final iteration of v2 before moving on to v3 API support (and will be branched as v2final).

### CODE REVIEW

File: BridgeSDK/Networking/SBBNetworkManager.m [hunk 1]
```diff
@@ -218,6 +218,11 @@ - (instancetype) initWithBaseURL: (NSString*) baseURL bridgeStudy: (NSString*)br
         self.environment = SBBEnvironmentCustom;
```
Comment (EricSiegNW) [inline BridgeSDK/Networking/SBBNetworkManager.m]: Is this right? Does it get reset somewhere else? I realize its not part of the PR changes but something I noticed

File: BridgeSDK/Networking/SBBNetworkManager.m [hunk 2]
```diff
@@ -218,6 +218,11 @@ - (instancetype) initWithBaseURL: (NSString*) baseURL bridgeStudy: (NSString*)br
...
         self.uploadCompletionHandlers = [NSMutableDictionary dictionary];
         self.downloadCompletionHandlers = [NSMutableDictionary dictionary];
+        if (bridgeStudy.length) {
```
Comment (EricSiegNW) [inline BridgeSDK/Networking/SBBNetworkManager.m]: Could you add a couple word comment explaining what is going on here?

File: BridgeSDKTests/SBBBridgeAPIUnitTestCase.m [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
 
-#import "SBBBridgeAPITestCase.h"
+#import "SBBBridgeAPIUnitTestCase.h"
 #import "SBBAuthManagerInternal.h"
 #import "SBBBridgeNetworkManager.h"
 #import "SBBNetworkManagerInternal.h"
+#import "TestAdminAuthDelegate.h"
 
-@implementation SBBBridgeAPITestCase
+@interface SBBBridgeAPIUnitTestCase ()
```
Comment (EricSiegNW) [inline BridgeSDKTests/SBBBridgeAPIUnitTestCase.m]: NIce differentiation, also excited to see integration tests :)

**Comment by EricSiegNW** [comment]
The code looks good. However, when I try to run the BridgeSDKIntegrationTests suite I get a build failure: The file "BridgeAdminCredentials.plist" couldn't be opened because there is no such file. Does that need to be checked in? I'm guessing it might be something that will exist in the CI environment rather than being checked in, but still wanted to call it out.

File: BridgeSDK/Networking/SBBNetworkManager.m (hunk 1 above)
Comment (Erin-Mounts) [inline BridgeSDK/Networking/SBBNetworkManager.m]: Yes, this is only a default setting. This initializer isn't exposed in the public API and callers reset it appropriately.

File: BridgeSDK/Networking/SBBNetworkManager.m (hunk 2 above)
Comment (Erin-Mounts) [inline BridgeSDK/Networking/SBBNetworkManager.m]: done

**Comment by Erin-Mounts** [comment]
Yes, as you might guess, that .plist contains admin account credentials for the integration tests to use, and when Alx sent them to me he suggested they were super sensitive and I should eat my computer after seeing them... so no, they won't be checked into a public Github repo any time soon... ;)

FYI the convention between Apple, Y Media, and Sage when reviewing PRs to any of the repos involved in ResearchKit (of which this is, peripherally, one) is for the reviewer to comment "r=me" when they're satisfied, before the PR is accepted.

**Comment by EricSiegNW** [comment]
r=me
